### PR TITLE
chore: deprecate `NgZorroAntdModule`

### PR DESCRIPTION
**Pull request by hsuanxyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[x] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by vthinkxie** [comment]
stackbliz 部分也要修改一下

**Comment by hsuanxyz** [comment]
> stackblitz 部分也要修改一下

https://github.com/NG-ZORRO/ng-zorro-antd/pull/4519/files#diff-94db1c6c42a66d905f5d02788dae028aR124

加了一个导出全部模块的文件，就和 Material 一样。

File: README.md [hunk 1]
```diff
@@ -79,13 +79,13 @@ $ npm install ng-zorro-antd
 
 ## 🔨 Usage
 
-Import the module into every module where you want to use the components.
+Import the component modules you want to use into your `app.module.ts` file or other lazy-loaded feature modules.
```
Comment (vthinkxie) [inline README.md]:
```suggestion
Import the component modules you want to use into your `app.module.ts` file and other feature modules.
```

File: components/ng-zorro-antd.module.ts [hunk 2]
```diff
@@ -204,6 +204,9 @@ export * from 'ng-zorro-antd/version';
     NzWaveModule
   ]
 })
+/**
+ * @deprecated Use secondary entry eg: `import { NzButtonModule } from 'ng-zorro-antd/button'`.
+ */
 export class NgZorroAntdModule {
```
Comment (vthinkxie) [inline components/ng-zorro-antd.module.ts]: add a warning in the constructor

File: docs/global-config.en-US.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ From version 8.3.0, we add support of **global configuration** to many component
 If you want to provide default configurations to some components, you should provide an object that implements the interface `NzConfig` with the injection token `NZ_CONFIG`, in the root module (in another word, to the root injector). Like this:
 
 ```typescript
-import { NgZorroAntdModule, NzConfig, NZ_CONFIG } from 'ng-zorro-antd';
+import { NzConfig, NZ_CONFIG } from 'ng-zorro-antd';
```
Comment (vthinkxie) [inline docs/global-config.en-US.md]:
```suggestion
import { NzConfig, NZ_CONFIG } from 'ng-zorro-antd/core';
```

File: docs/global-config.zh-CN.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ title: 全局配置项
 想要为某些组件提供默认配置项，请在根注入器中根据注入令牌 `NZ_CONFIG` 提供一个符合 `NzConfig` 接口的对象，例如：
 
 ```typescript
-import { NgZorroAntdModule, NzConfig, NZ_CONFIG } from 'ng-zorro-antd';
+import { NzConfig, NZ_CONFIG } from 'ng-zorro-antd';
```
Comment (vthinkxie) [inline docs/global-config.zh-CN.md]:
```suggestion
import { NzConfig, NZ_CONFIG } from 'ng-zorro-antd/core';
```

**Comment by vthinkxie** [comment]
内部组件加 ɵ

**Comment by yinuocode** [comment]
为何要分开导入模块呢，之前的不是也是有摇树优化的吗

**Comment by vthinkxie** [comment]
@yinuocode 分开导入模块可以提升编译速度，与摇树优化没有关系

**Pull request merged**